src/test/java/core/basesyntax/service/FruitServiceImplTest.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.TestObjects.APPLE;
+import static core.basesyntax.TestObjects.BANANA;
+import static core.basesyntax.TestObjects.FIRST_LINE;
+import static core.basesyntax.TestObjects.HEADER;
+import static core.basesyntax.TestObjects.SECOND_LINE;
+import static core.basesyntax.model.FruitTransaction.Operation.BALANCE;
+import static core.basesyntax.model.FruitTransaction.Operation.SUPPLY;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.repository.Storage;
+import core.basesyntax.service.parser.CsvParser;
+import core.basesyntax.service.reader.CsvReader;
+import core.basesyntax.service.report.ReportGenerator;
+import core.basesyntax.service.storage.StorageService;
+import core.basesyntax.service.writer.CsvWriter;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitServiceImplTest {
+    private CsvReader csvReader;
+    private CsvParser csvParser;
+    private StorageService storageService;
+    private Storage storage;
+    private ReportGenerator reportGenerator;
+    private CsvWriter csvWriter;
+    private FruitServiceImpl fruitService;
+
+    @BeforeEach
+    void setUp() {
+        csvReader = mock(CsvReader.class);
+        csvParser = mock(CsvParser.class);
+        storageService = mock(StorageService.class);
+        storage = mock(Storage.class);
+        reportGenerator = mock(ReportGenerator.class);
+        csvWriter = mock(CsvWriter.class);
+        fruitService = new FruitServiceImpl(csvReader, csvParser, storageService,
+                storage, reportGenerator, csvWriter);
```

Comment:
It is very nice that you used mocks, good job! But there are simpler ways to define them, you can just add @Mock annotation to classes that you want to mock. But you can leave as is